### DISCUSSION TITLE
Native HTML popover API

### DISCUSSION THREAD

**Discussion by laurentpayot** [beercss/beercss#187] [Ideas]
[The popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API) will be [available in Safari 17](https://www.webkit.org/blog/14205/news-from-wwdc23-webkit-features-in-safari-17-beta/#popover). It is [already available on Chrome/Edge](https://caniuse.com/mdn-api_htmlelement_popover) (under a flag in Firefox).

It can be used for (at least):
- dialogs (closed when clicked outside, just like MD3 dialogs)
- menus
- tooltips
- ~~toasts~~ snackbars

A great resource is https://developer.chrome.com/blog/introducing-popover-api.

What are your thought about integrating popovers to beercss when it is a bit more widespread?

**Comment by laurentpayot** [comment]
The popover API is available in recent versions of Chrome and Safari, and [Firefox 124 will support it](https://caniuse.com/mdn-api_htmlelement_popover). Just sayin’... :wink:

**Comment by coffeeandwork** [comment]
Very cool.

**Comment by laurentpayot** [comment]
[The Popover API is now Baseline Newly available](https://web.dev/blog/popover-baseline) :tada:
